server/src/cloudify/rest.ts

```diff
@@ -0,0 +1,71 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Cloudify Platform LTD. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+
+import fetch from 'node-fetch';
+
+// For casting the response of our Marketplace API.
+interface JSONPagination {
+	'size': number,
+	'offset': number,
+	'total': number,
+}
+export interface JSONItems {
+    [key: string]: Object;
+}
+interface JSONResponse {
+    items: JSONItems;
+	pagination: JSONPagination;
+};
```

Comment:
In the future you can make it more generic in the following way:
```suggestion
export interface JSONItems<ItemType> {
    [key: string]: ItemType;
}
interface JSONResponse<ItemType> {
    items: JSONItems<ItemType>;
	pagination: JSONPagination;
};
```

then instead of
```
export function rawRequest(url:string, method:string, additionalParams={}): Promise<JSONResponse> {
```
you could have:
```
export function rawRequest<ItemType>(url:string, method:string, additionalParams={}): Promise<JSONResponse<ItemType>> {
```

and knowing what kind of item you are expecting you would be able to call `rawRequest<MyItemType>(...)` and receive Promise responses with more accurate type (instead of generic object).

Reply:
This is good stuff. I also don't need the paginated reponse fn anymore. So I'm dumping it.